src/components/forms/price/PriceForm.tsx

```diff
@@ -145,12 +145,17 @@ async function onSubmit(
   const price = Price.create({
     ...data,
     date: DateTime.fromISO(data.date),
-    guid: data.guid || undefined,
+    guid: undefined,
   });
 
-  if (action === 'add' || action === 'update') {
+  if (action === 'add') {
     await price.save();
+  } else if (action === 'update') {
+    await price.save();
+    price.guid = data.guid;
+    await price.remove();
```

Comment:
Discovered a bug where updating the date of an existing Price was failing because it said duplicated key. Seems upsert functionality is trying to add a new price when one of the elements in the unique key constraint changes. This is a bit confusing. Don't know if there is a better way but for now the approach I'm doing is to save without `guid` (so a new instance is added` and then delete the previous one